Wrappers/Python/cil/optimisation/algorithms/GD.py

```diff
@@ -40,27 +40,28 @@ class GD(Algorithm):
     preconditioner: class with a `apply` method or a function that takes an initialised CIL function as an argument and modifies a provided `gradient`.
             This could be a custom `preconditioner` or one provided in :meth:`~cil.optimisation.utilities.preconditioner`. If None is passed  then `self.gradient_update` will remain unmodified. 
 
-    rtol: positive float, default 1e-5
-        optional parameter defining the relative tolerance comparing the current objective function to 0, default 1e-5, see numpy.isclose
-    atol: positive float, default 1e-8
-        optional parameter defining the absolute tolerance comparing the current objective function to 0, default 1e-8, see numpy.isclose
-
     """
 
-    def __init__(self, initial=None, objective_function=None, step_size=None, rtol=1e-5, atol=1e-8,  preconditioner=None, **kwargs):
+    def __init__(self, initial=None, objective_function=None, step_size=None,   preconditioner=None, **kwargs):
         '''GD algorithm creator
         '''
 
         self.alpha = kwargs.pop('alpha', None)
         self.beta = kwargs.pop('beta', None)
-
+        self.rtol = kwargs.pop('rtol', 0) # to be deprecated
+        self.atol = kwargs.pop('atol', 0) # to be deprecated
+        
         super().__init__(**kwargs)
 
         if self.alpha is not None or self.beta is not None:
             warn('To modify the parameters for the Armijo rule please use `step_size_rule=ArmijoStepSizeRule(alpha, beta, kmax)`. The arguments `alpha` and `beta` will be deprecated. ', DeprecationWarning, stacklevel=2)
 
-        self.rtol = rtol
-        self.atol = atol
+
+        if self.rtol!=0 or self.atol!=0: # to be deprecated
+            warn('`rtol` and `atol` are deprecated. For early stopping, please use a callback (cil.optimisation.utilities.callbacks) instead', DeprecationWarning, stacklevel=2)
```

Comment:
There is no example callback we can use?

Reply:
I gave an example